packages/flutter/lib/src/gestures/events.dart

```diff
@@ -790,6 +790,61 @@ class PointerUpEvent extends PointerEvent {
   );
 }
 
+/// A base type for events that correspond to a discrete pointer signal.
+///
+/// There are events that originate from the pointer but don't change the state
+/// of the pointer itself, and are discrete rather than needing to be
+/// interpreted in the context of a series of events.
+abstract class PointerSignalEvent extends PointerEvent {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const PointerSignalEvent({
+    Duration timeStamp = Duration.zero,
+    int pointer = 0,
+    PointerDeviceKind kind = PointerDeviceKind.mouse,
+    int device = 0,
+    Offset position = Offset.zero,
+  }) : super(
+    timeStamp: timeStamp,
+    pointer: pointer,
+    kind: kind,
+    device: device,
+    position: position,
+  );
+}
+
+/// The pointer issued a scroll event.
+///
+/// For example, the scroll wheel on a mouse was scrolled.
+class PointerScrollEvent extends PointerSignalEvent {
+  /// Creates a pointer scroll event.
+  ///
+  /// All of the arguments must be non-null.
```

Comment:
If this is true, then there should be asserts to enforce it.

Reply:
Done. One of the asserts fired after I added it, which made me realize that `pointer` shouldn't actually be a field on a scroll event, since a pointer in the Flutter event system corresponds to a touch, essentially, and scrolls don't have to (and usually don't) come from a pointer that's down, nor does the scroll system need to know what pointer if any it's associated with. Yay for assertions :)

Related: all of the pointer event classes say that everything must be non-null, but none of them assert; would you like me to file a bug to track the missing asserts? (I'm not sure if that would be considered breaking, and thus potentially not worth doing.)

Reply:
Yes, please file the issue.  We should be enforcing contracts about things not being null.  At least until we get non-nullable types. :-)

By definition, if we break someone by adding an assert to enforce an existing contract, then we just helped them find an error in their code.